File: semester_two/algoDat/public/code/vorlesung-10/vorlagen/java/GraphTest.java
import java.io.IOException;
import java.net.URL;

import org.junit.Assert;
import org.junit.Test;

/**
 * Test class for our graph.
 * 
 * @author Claudius Korzen.
 * @author Axel Lehmann
 */
public class GraphTest {
  /** Test the constructor. */
  @Test
  public void testConstructor() {
    Graph graph = new Graph();
    Assert.assertEquals(0, graph.numNodes);
    Assert.assertEquals(0, graph.numEdges);
    Assert.assertEquals(0, graph.adjacencyLists.size());
    Assert.assertEquals(0, graph.nodes.size());
  }

  /** Test the read method. */
  @Test
  public void testRead() throws IOException {
    Graph graph = new Graph();
    URL url = ClassLoader.getSystemResource("example.graph");
    if (url != null) {
      graph.read(url.getPath());
      Assert.assertEquals(5, graph.numEdges);
      Assert.assertEquals(4, graph.numNodes);
      Assert.assertEquals("[0->3|3, 1->0|1, 1->2|-2, 2->3|2, 3->2|5]", 
          graph.toString());
    }
  }
}

File: semester_two/algoDat/own/blatt-10/GraphTest.cpp
// Copyright 2017, University of Freiburg,
// Chair of Algorithms and Data Structures.
// Author: Hannah Bast <bast@cs.uni-freiburg.de>,
//         Axel Lehmann <lehmann@cs.uni-freiburg.de>.

#include <gtest/gtest.h>
#include "./Graph.h"

// _____________________________________________________________________________
TEST(GraphTest, constructor) {
  Graph graph;
  ASSERT_EQ("[]", graph.toString());
}

// _____________________________________________________________________________
TEST(GraphTest, read) {
  Graph graph;
  graph.read("example.graph");
  ASSERT_EQ("[0->3|3, 1->0|1, 1->2|-2, 2->3|2, 3->2|5]", graph.toString());
}
